providers/src/airflow/providers/databricks/hooks/databricks_base.py

```diff
@@ -678,6 +679,9 @@ def _retryable_error(exception: BaseException) -> bool:
             if exception.status >= 500 or exception.status == 429:
                 return True
 
+        if isinstance(exception, ClientConnectorError):
```

Comment:
I did not add any status check here, as the connection was not even established. Hence, there's no status to be checked